zerver/lib/remote_server.py

```diff
@@ -226,6 +229,16 @@ def send_to_push_bouncer(
             # The callers requesting this endpoint want the exception to propagate
             # so they can catch it.
             raise RemoteRealmServerMismatchError
+        elif endpoint == "push/e2ee/register" and "code" in result_dict:  # nocoverage
```

Comment:
This function does a ton of `if "code" in result_dict and result_dict["code"] == "INVALID_ZULIP_SERVER"`.

We should just make a variable `code` outside that grouping, so `code=None` if not present, and otherwise it's whatever it is.

Reply:
Added a commit to simplify this. (2nd last commit)